### DISCUSSION TITLE
How to apply useGroupBy for both rows and column?

### DISCUSSION THREAD

**Discussion by sangdth** [TanStack/table#2004] [General]
I'm really interested in react-table and want to convince my team to adopt it. For now there is one problem I've encountered, which is how to apply the groupBy for both row and column.

The product I'm working on is somewhat similar to Excel, allow users to create a pivot table, then create a matrix table or chart from that data.

I've checked the examples and API doc, but it seems I cannot find a place to make the group by column at the same time with the group by row, I hope I miss something. Please help.

Thanks.

---

**Update 1:** 
Seems like I have 2 options, still not sure. One is using recursive header columns, the other way is `togglePivot`. How can I find the details `usePivotColumns` documentation?

**Comment by wanhchai** [comment]
Hi, Im interested to see your solution as well.
